File: code_combat/7_Sarven_Desert/261-The_Dunes/the_dunes.py
while True:
    enemy = hero.findNearestEnemy()
    if enemy:
        if enemy.type != "sand-yak" and enemy.type != "burl":
            distance = hero.distanceTo(enemy)
            if distance < 45:
                if distance <= 15 and hero.canCast("drain-life"):
                    hero.cast("drain-life", enemy)
                else
                    hero.attack(enemy)

    item = hero.findNearestItem()
    if item:
        hero.moveXY(item.pos.x, item.pos.y)
    else:
        hero.moveXY(41, 31)

File: projects/code_combat/7_Sarven_Desert/261-The_Dunes/theDunes.js
// Collect coins. Ignore "sand-yak"s and "burl"s.

while (true) {
    var enemy = hero.findNearestEnemy();
    var item = hero.findNearestItem();
    if (enemy) {
        if (enemy.type == "sand-yak" || enemy.type == "burl") {
            // Don't attack! Collect coins.
            if (item) {
                hero.moveXY(item.pos.x, item.pos.y);
            }
        } else {
            // Else, attack. 
            hero.attack(enemy);
        }
    } else if (item) {
        // Collect coins: move to item's position.
        hero.moveXY(item.pos.x, item.pos.y);
    } else {
        hero.moveXY(41, 31);
    }
}
